src/geo/projection/globe_util.js

```diff
@@ -395,15 +396,27 @@ export function globePoleMatrixForTile(z: number, x: number, tr: Transform): Flo
     return Float32Array.from(poleMatrix);
 }
 
-export function getGridMatrix(id: CanonicalTileID, corners: Array<Array<number>>): Array<number> {
+export function getGridMatrix(id: CanonicalTileID, corners: Array<Array<number>>, latitudinalLod: number): Array<number> {
     const [tl, br] = corners;
     const S = 1.0 / GLOBE_VERTEX_GRID_SIZE;
     const x = (br[1] - tl[1]) * S;
-    const y = (br[0] - tl[0]) * S;
+    const latitudinalSubdivs = GLOBE_LATITUDINAL_GRID_LOD_TABLE[latitudinalLod];
+    const y = (br[0] - tl[0]) / latitudinalSubdivs;
     const tileZoom = 1 << id.z;
     return [0, x, tileZoom, y, 0, id.y, tl[0], tl[1], S];
 }
 
+export function getLatitudinalLod(id: CanonicalTileID, corners: Array<Array<number>>): number {
+    const UPPER_LATITUDE = MAX_MERCATOR_LATITUDE - 5.0;
+    const [tl, br] = corners;
+    let lat = (tl[0] + br[0]) / 2.0;
+    lat = clamp(lat, -UPPER_LATITUDE, UPPER_LATITUDE) / UPPER_LATITUDE * 90.0;
+    // const t = Math.pow(1.0 - Math.cos(degToRad(lat)), 2);
+    const t = Math.pow(Math.abs(Math.sin(degToRad(lat))), 3);
+    const lod = Math.round(t * (GLOBE_LATITUDINAL_GRID_LOD_TABLE.length - 1));
+    return lod;
+}
```

Comment:
Could this be unit tested to help understand the input/output behavior?

Reply:
> Over how many frames and which platform was this calculated? Also feel free to share the diff you used so we can try running that on different machines.

@karimnaaji It was on Chrome on a Linux machine. I have included the code (in `flytolocations.html`)